tools/New-AzureSampleResources.ps1

```diff
@@ -0,0 +1,169 @@
+[System.Environment]::SetEnvironmentVariable('AZURE_SP_CERT_OBJECT_ID', $azureCertSpInfo["ObjectId"], [System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('AZURE_SP_CERT_PATH', $certPfxPath, [System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('AZURE_SP_CERT_PASS', $certPfxPassword, [System.EnvironmentVariableTarget]::Machine)
+
+[System.Environment]::SetEnvironmentVariable('AZURE_SUBSCRIPTION_ID', $AzureSubscriptionId, [System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('AZURE_ARM_ENDPOINT', $azureAdminArmEndpoint, [System.EnvironmentVariableTarget]::Machine)
+[System.Environment]::SetEnvironmentVariable('AZURE_LOCATION', $azureLocation, [System.EnvironmentVariableTarget]::Machine)
+
+######################
+# SDK SPECIFIC SETUP
+######################
+
+switch ($env:PROCESSOR_ARCHITECTURE) 
+{
+    "AMD64" { $architecture = "x64" }
+    "x86" { $architecture = "x86" }
+    default { throw "PowerShell package for OS architecture '$_' is not supported." }
+}
+
+$logFolder = $([io.path]::combine($env:HOMEPATH, "AzureSampleLogs"))
+If (-not (Test-Path -Path $logFolder))
+{
+    New-Item $logFolder -Type Directory -ErrorAction Stop
+}
+
+if ($NodeJS)
+{
+    # Install NodeJS
+    $nodeJSVersion = "v14.17.0"
+    $nodeJSDownloadURL = "https://nodejs.org/dist/${nodeJSVersion}/node-${nodeJSVersion}-${architecture}.msi"
+    $nodeJSInstallLogFilePath = Join-Path -Path $logFolder -ChildPath "nodejsInstall.log"
+    Install-SoftwareFromURL -DownloadURL $nodeJSDownloadURL `
+        -RegistryDisplayNameLike "Node.js" `
+        -InstallLogFilePath $nodeJSInstallLogFilePath `
+        -ForceInstall
+
+    # Set up Azure Stack local certificate for NodeJS.
+    $certFolder = [System.IO.Path]::Combine($env:HOMEPATH, "certs")
+    if (!(Test-Path -Path $certFolder))
+    {
+        New-Item -ItemType "directory" -Path $certFolder
+    }
+
+    New-NodeJSEnvPem -PemFolder $certFolder
+}
+
+if ($Golang)
+{
+    # Install Golang
+    # To get the latest Golang version use:
+    #     $wc = New-Object System.Net.WebClient
+    #     [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+    #     $latestVersion = $wc.DownloadString("https://golang.org/VERSION?m=text")
+    # WARNING: Golang updates usually have breaking changes, becareful when updating.
+    $golangVersion = "go1.15.8"
+    $downloadFileName = "${golangVersion}.windows-amd64.msi"
+    $golangDownloadURL = "https://golang.org/dl/${downloadFileName}"
+    $GoSDKInstallLogFilePath = Join-Path -Path $logFolder -ChildPath "goInstall.log"
+    Install-SoftwareFromURL -DownloadURL $golangDownloadURL `
+        -RegistryDisplayNameLike "Go Programming Language*" `
+        -InstallLogFilePath $GoSDKInstallLogFilePath `
+        -ForceInstall
+
```

Comment:
do we need to whitelist the AZS cert for go and nodejs store? if we need, it's probably good idea to do it as part of this script itself.

Reply:
What do you mean by whitelist? The Azs cert is expected to work without in the environments already because CI tests use them.